sway-lib-std/src/u256.sw

```diff
@@ -0,0 +1,89 @@
+library u256;
+
+use core::num::*;
+use ::result::Result;
+
+/// The 256-bit unsigned integer type.
+/// Represented as four 64-bit components: `(a, b, c, d)`, where `value = (a << 192) + (b << 128) + (c << 64) + d`.
+pub struct U256 {
+    a: u64,
+    b: u64,
+    c: u64,
+    d: u64,
+}
+
+pub enum U256Error {
+    LossOfPrecision: (),
+}
+
+pub trait From {
+    /// Function for creating a U256 from its u64 components.
+    pub fn from(a: u64, b: u64, c: u64, d: u64) -> Self;
+} {
+    /// Function for extracting 4 u64s from a U256.
+    fn into(val: U256) -> (u64, u64, u64, u64) {
+        (val.a, val.b, val.c, val.d)
+    }
+}
+
+impl From for U256 {
+    pub fn from(a: u64, b: u64, c: u64, d: u64) -> U256 {
+        U256 {
+            a, b, c, d,
+        }
+    }
+}
+
+impl core::ops::Eq for U256 {
+    /// Function for comparing 2 U256s for equality
+    pub fn eq(self, other: Self) -> bool {
+        self.a == other.a && self.b == other.b && self.c == other.c && self.d == other.d
+    }
+}
+
+impl U256 {
+    /// Initializes a new, zeroed U256.
+    pub fn new() -> U256 {
+        U256 {
+            a: 0,
+            b: 0,
+            c: 0,
+            d: 0,
+        }
+    }
+
+    /// Downcast to `u64`. Err if precision would be lost, Ok otherwise.
+    pub fn to_u64(self) -> Result<u64, U256Error> {
```

Comment:
Do we want this to be like https://docs.rs/primitive-types/0.6.1/primitive_types/struct.U256.html#method.as_u64? Same as: https://docs.rs/bigint/4.2.0/bigint/uint/struct.U256.html#method.as_u64

Reply:
Yes, it makes sense to rename `to_u64()` => `as_u64()` here.
This sets us up for implementing more methods in line with what rust does and following the same naming convention.